### ISSUE TITLE
csharp_grpc_compile assumes that all dependent protos have services

### ISSUE THREAD

**Issue by amlinux** [rules-proto-grpc/rules_proto_grpc#48]
If a proto file with a service contains a dependency on another proto file **without** services, `csharp_grpc_compile` fails with an error:

```
ERROR: /home/aml/tmp/rules-proto-grpc-bug/BUILD.bazel:3:1: output 'mypackage_proto/csharp_grpc_compile_aspect_verb0/MypackageGrpc.cs' was not created
ERROR: /home/aml/tmp/rules-proto-grpc-bug/BUILD.bazel:3:1: not all outputs were created or valid
```

Min repro: https://github.com/amlinux/rules-proto-grpc-bug

**Comment by bnczk** [comment]
This is the same for the grpc-web rules

**Comment by amlinux** [comment]
Actually I think the root cause is that each rule invocation tries to generate sources for all transitively dependent protos. This not only leads to the problem described here, but also doesn't allow diamond dependencies.

A -> B -> C
A -> D -> C

C just gets generated twice, and linking fails.

**Comment by bnczk** [comment]
I don't think this is an issue with dependencies in general. In you repro repo, when changing the `myservice_grpc_csharp` rule to use `:mypackage_proto`, your problem appears again. 

AFAIU, the problem is that the csharp grpc proto_plugin expects the outputs to be `"{basename|pascal}Grpc.cs"` for each proto file. But if no service is in the proto file, no corresponding `"{basename|pascal}Grpc.cs"` will be generated. Replacing the `outputs` with `output_directory = True` fixes it for me in your repro repo.

**Comment by amlinux** [comment]
I should have said it differently: there is a separate issue with diamond dependencies, and once it gets fixed (likely by not generating transitive dependencies), this one will go away automatically.

**Comment by aaliddell** [comment]
Fixed in #84
